### PR TITLE
GH-168 remove python 3.8 from tests

### LINKED ISSUE #168: remove python 3.8 from ci workflows
Remove python version 3.8 from the CI workflows to match romancal

### PR DESCRIPTION
**Pull request by ddavis-stsci**
Resolves [RCAL-nnnn](https://jira.stsci.edu/browse/RCAL-nnnn)

Closes #168

This PR removes python 3.8 from the ci tests and adds 3.11 where needed. 

Note: the documentation error has a separate ticket,  #[rcal-555](https://jira.stsci.edu/projects/RCAL/issues/RCAL-555) 

Unit tests: 1226 passed, 1 skipped in 99.35s

The regression tests are at: 
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/169/

**Checklist**
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [x] updated relevant documentation
- [x] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/XXX/

`pytest tests/
=================================================================== test session starts ===================================================================
platform darwin -- Python 3.11.0, pytest-7.3.1, pluggy-1.0.0
rootdir: /Users/ddavis/src/Roman/roman_datamodels
configfile: pyproject.toml
plugins: remotedata-0.4.0, hypothesis-6.75.1, asdf-2.15.0, mock-3.10.0, filter-subpackage-0.1.2, astropy-header-0.2.2, doctestplus-0.12.1, astropy-0.10.0, cov-4.0.0, ci-watson-0.6.1, openfiles-0.5.0, arraydiff-0.5.0
collected 1227 items

tests/test_factories.py ............................................................................................s                               [  7%]
tests/test_filetype.py .                                                                                                                            [  7%]
tests/test_models.py ...................................................                                                                            [ 11%]
tests/test_open.py ........                                                                                                                         [ 12%]
tests/test_random_utils.py ........................................................................................................................ [ 22%]
................................................................................................................................................... [ 34%]
................................................................................................................................................... [ 46%]
................................................................................................................................................... [ 58%]
................................................................................................................................................... [ 70%]
................................................................................................................................................... [ 82%]
................................................................................................................................................... [ 94%]
......                                                                                                                                              [ 94%]
tests/test_stnode.py ..................................................                                                                             [ 98%]
tests/test_stuserdict.py ....                                                                                                                       [ 99%]
tests/test_units.py ............                                                                                                                    [100%]

======================================================= 1226 passed, 1 skipped in 99.35s`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,10 +26,6 @@ jobs:
     uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
     with:
       envs: |
-        - linux: test-oldestdeps-cov-xdist
-          python-version: 3.8
-        - linux: test-xdist
-          python-version: 3.8
         - linux: test-xdist
```
Comment (zacharyburnett) [inline .github/workflows/ci.yml:29]:
```suggestion
        - linux: test-oldestdeps-cov-xdist
          python-version: 3.9
        - linux: test-xdist
```

File: .github/workflows/ci_cron.yml [hunk 2]
```diff
@@ -11,10 +11,12 @@ jobs:
...
         - macos: test-xdist
           python-version: 3.9
         - macos: test-xdist
           python-version: 3.10
-        - linux: test-rad-xdist
+        - linux: test-xdist
+          python-version: 3.11
+        - macos: test-xdist
+          python-version: 3.11
+         - linux: test-rad-xdist
```
[the remark this answers is not in the document]
Comment (ddavis-stsci) [reply .github/workflows/ci_cron.yml]: Fixed

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ddavis-stsci) [reply .github/workflows/ci.yml]: Thanks - it should be updated.

**Comment by zacharyburnett** [comment]
It looks like the docs build is failing, due to the `.readthedocs.yaml` file using Python 3.8. 

The `python.version` entry is now deprecated, so newer versions of Python need to use the `build.tools.python` entry it can be; either `3.11` or `mambaforge`: https://docs.readthedocs.io/en/stable/config-file/v2.html#build-tools-python

Here is the one for the JWST pipeline:
https://github.com/spacetelescope/jwst/blob/767ed34117e60730ce832eb4ee430f79cb8a76e5/.readthedocs.yaml#L19-L34

**Comment by zacharyburnett** [comment]
Oh I apologize, you will also need an `environment.yaml` file with that configuration: https://github.com/spacetelescope/jwst/blob/767ed34117e60730ce832eb4ee430f79cb8a76e5/docs/rtd_environment.yaml

**Comment by ddavis-stsci** [comment]
NP, It is all working correctly now, Thanks.

**Comment by WilliamJamieson** [comment]
This PR is incomplete, it should also drop python 3.8 support in the `pyproject.toml` and strip any python < 3.9 code. If we drop testing, we will no-longer be able to determine if any bugs on that python version appear. Therefore we should indicate that to end users and drop python 3.8 specific support.

**Pull request merged**